src/rust/src/x509/certificate.rs

```diff
@@ -319,6 +319,98 @@ impl Certificate {
             },
         )
     }
+
+    fn verify_signed_by<'p>(
+        &self,
+        py: pyo3::Python<'p>,
+        issuer: pyo3::PyRef<'_, Certificate>,
+    ) -> pyo3::PyResult<&'p pyo3::PyAny> {
+        let public_key = issuer.public_key(py)?;
```

Comment:
I think the core of this logic should go in `sign.rs` for symmetry -- it can be a method that takes the public key, tbs, and signature bytes.